dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java

```diff
@@ -607,6 +607,7 @@ private ImportSummary addEnrollment( Enrollment enrollment, ImportOptions import
         }
 
         programInstance.setCreatedByUserInfo( UserInfoSnapshot.from( importOptions.getUser() ) );
+        programInstance.setLastUpdatedByUserInfo( UserInfoSnapshot.from( importOptions.getUser() ) );
```

Comment:
What is the meaning of the `lastUpdated` columns? Is this method creating an enrollment?

I would not expect `lastUpdated` to be populated when something was never updated.

Reply:
At creation, we set lastUpdated = createdAt, although there have been no change from the original (as it doesn't exist any more) the lastUpdated also represent when it was last worked on, which is initially the same as createdAt. This looks good to me :)

Reply:
Im not sure I see where we add the lastUpdatedBy object when just updating an enrollment? I see events -> enrollments, and create enrollment. Could you tell us where you covered the case of updating (Lets say attribute values of an enrollment) @lucaCambi77 ?

Reply:
This is change is requested by Calle Hedberg. Every time enrollments get updated, the `lastUpdatedByUserInfo` should be set. From the enrollment page, we can add or update events and this will be a call to /events endpoint.
So I am wondering now if updating  the `lastUpdatedByUserInfo` for the `programinstance` table makes sense then if we update the events

Reply:
So, to clarify, here are the cases we should update the lastUpdatedByUserInfo:

1. Create new enrollment
2. Update enrollment status, attributes, etc
3. Create event in enrollment
4. update event in enrollment

We only add createByUserInfo when creating the enrollment, which is already done.

Of those 4 cases over, I think this PR only covers 1, 3 and 4. So we are missing 1 case.